tls/s2n_alerts.c

```diff
@@ -315,10 +259,62 @@ S2N_RESULT s2n_queue_reader_no_renegotiation_alert(struct s2n_connection *conn)
      *# handshake_failure alert.
      **/
     if (s2n_connection_get_protocol_version(conn) == S2N_SSLv3) {
-        RESULT_GUARD_POSIX(s2n_queue_reader_alert(conn, S2N_TLS_ALERT_LEVEL_FATAL, S2N_TLS_ALERT_HANDSHAKE_FAILURE));
+        RESULT_GUARD_POSIX(s2n_queue_reader_handshake_failure_alert(conn));
+        RESULT_BAIL(S2N_ERR_BAD_MESSAGE);
+    }
+
+    if (!conn->reader_warning_out) {
+        conn->reader_warning_out = S2N_TLS_ALERT_NO_RENEGOTIATION;
+    }
+    return S2N_RESULT_OK;
+}
+
+S2N_RESULT s2n_alerts_write_error_or_close_notify(struct s2n_connection *conn)
```

Comment:
```suggestion
S2N_RESULT s2n_alerts_write_fatal(struct s2n_connection *conn)
```
Shouldn't the name mirror 's2n_alerts_write_warning()'?

Reply:
Maybe, but it feels wrong to call "close_notify" fatal, even though we sort of use it that way. I prefer the more descriptive name, especially since the RFC I quote right before calling this says "Each party MUST send a "**close_notify**" alert before closing its write side of the connection, unless it has already sent some **error alert**."